fixes for blackmailer issue and hd64180

cm68 opened the pull request:
added font builder, compressor, 64180 support, huge speedup for copy
fixed assembler bug with embedded quoted double quote

b1ackmai1er commented:
Keen to try this out - thats a lot of work - well done - I noticed yesterday that hbios.asm had some assembly errors due to embedded quote with the floppy description  PS_FLP_FSTR table. I don't think that's been identified in this commit. Apologies if I'm wrong.

cm68 commented:
this is fixed in the pull request i just sent.


On Sat, Feb 22, 2020 at 11:25 PM b1ackmai1er <notifications@github.com>
wrote:

> Keen to try this out - thats a lot of work - well done - I noticed
> yesterday that hbios.asm had some assembly errors due to embedded quote
> with the floppy description PS_FLP_FSTR table. I don't think that's been
> identified in this commit. Apologies if I'm wrong.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wwarthen/RomWBW/pull/85?email_source=notifications&email_token=AJJUTK5UD3WKS6ZTMTZPDEDREIQFPA5CNFSM4KZYD7DKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMVTT7A#issuecomment-590035452>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJJUTK7C7QK7TUONXEYPWFTREIQFPANCNFSM4KZYD7DA>
> .
>


-- 
*I *refine, *they* debug, *he/she* patches,* they* kludge.

wrljet commented:
I gave this a test drive on Ubuntu 19.10 x64.

While running `make diff`, I received a bunch of these errors: `/bin/sh: 4: [: -a: unexpected operator`

I had unzipped `RomWBW-2.9.2-pre.32-Package.zip` into `~/RomWBW.windows` to compare with.

`bill@ubuntu:~/RomWBW-cm68$ make diff`
`cd Source ; make diff`
`make[1]: Entering directory '/home/bill/RomWBW-cm68/Source'`
`diff in /home/bill/RomWBW-cm68/Source/Prop`
`make[2]: Entering directory '/home/bill/RomWBW-cm68/Source/Prop'`
`diff in /home/bill/RomWBW-cm68/Source/Prop/Spin`
`make[3]: Entering directory '/home/bill/RomWBW-cm68/Source/Prop/Spin'`
`make[3]: Leaving directory '/home/bill/RomWBW-cm68/Source/Prop/Spin'`
`make[2]: Leaving directory '/home/bill/RomWBW-cm68/Source/Prop'`
`diff in /home/bill/RomWBW-cm68/Source/Apps`
`make[2]: Entering directory '/home/bill/RomWBW-cm68/Source/Apps'`
`diff in /home/bill/RomWBW-cm68/Source/Apps/XM`
`make[3]: Entering directory '/home/bill/RomWBW-cm68/Source/Apps/XM'`
`/bin/sh: 4: [: -a: unexpected operator`
`/bin/sh: 4: [: -a: unexpected operator`
`make[3]: Leaving directory '/home/bill/RomWBW-cm68/Source/Apps/XM'`
`diff in /home/bill/RomWBW-cm68/Source/Apps/FDU`
`make[3]: Entering directory '/home/bill/RomWBW-cm68/Source/Apps/FDU'`
`/bin/sh: 4: [: -a: unexpected operator`

Bill

cm68 commented:
hi.  thanks.  I'll see what's going on here.

--curt

On Sun, Feb 23, 2020 at 8:07 AM Bill Lewis <notifications@github.com> wrote:

> I gave this a test drive on Ubuntu 19.10 x64.
>
> While running make diff, I received a bunch of these errors: /bin/sh: 4:
> [: -a: unexpected operator
>
> I had unzipped RomWBW-2.9.2-pre.32-Package.zip into ~/RomWBW.windows to
> compare with.
>
> bill@ubuntu:~/RomWBW-cm68$ make diff
> cd Source ; make diff
> make[1]: Entering directory '/home/bill/RomWBW-cm68/Source'
> diff in /home/bill/RomWBW-cm68/Source/Prop
> make[2]: Entering directory '/home/bill/RomWBW-cm68/Source/Prop'
> diff in /home/bill/RomWBW-cm68/Source/Prop/Spin
> make[3]: Entering directory '/home/bill/RomWBW-cm68/Source/Prop/Spin'
> make[3]: Leaving directory '/home/bill/RomWBW-cm68/Source/Prop/Spin'
> make[2]: Leaving directory '/home/bill/RomWBW-cm68/Source/Prop'
> diff in /home/bill/RomWBW-cm68/Source/Apps
> make[2]: Entering directory '/home/bill/RomWBW-cm68/Source/Apps'
> diff in /home/bill/RomWBW-cm68/Source/Apps/XM
> make[3]: Entering directory '/home/bill/RomWBW-cm68/Source/Apps/XM'
> /bin/sh: 4: [: -a: unexpected operator
> /bin/sh: 4: [: -a: unexpected operator
> make[3]: Leaving directory '/home/bill/RomWBW-cm68/Source/Apps/XM'
> diff in /home/bill/RomWBW-cm68/Source/Apps/FDU
> make[3]: Entering directory '/home/bill/RomWBW-cm68/Source/Apps/FDU'
> /bin/sh: 4: [: -a: unexpected operator
>
> Bill
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wwarthen/RomWBW/pull/85?email_source=notifications&email_token=AJJUTK3WIM46OUIHXQVKNL3REKNMXA5CNFSM4KZYD7DKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMV7UPA#issuecomment-590084668>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJJUTKYRWXMRRNAGCFG5UP3REKNMXANCNFSM4KZYD7DA>
> .
>


-- 
*I *refine, *they* debug, *he/she* patches,* they* kludge.

wwarthen commented:
Thanks Curt.  Since the changes should not affect existing Windows build, I am going to merge this and then do some testing.

cm68 commented:
thanks.  I'm doing more regression testing as well.

--curt

On Sun, Feb 23, 2020 at 12:52 PM Wayne Warthen <notifications@github.com>
wrote:

> Thanks Curt. Since the changes should not affect existing Windows build, I
> am going to merge this and then do some testing.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wwarthen/RomWBW/pull/85?email_source=notifications&email_token=AJJUTK5RSLH7GFSWAGZDX7LRELOX3A5CNFSM4KZYD7DKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMWG65A#issuecomment-590114676>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJJUTK4OUHIUVJ4XBBG7NU3RELOX3ANCNFSM4KZYD7DA>
> .
>


-- 
*I *refine, *they* debug, *he/she* patches,* they* kludge.

wwarthen commented:
Hi Curt,

My testing has gone very well.  I have created a some bit-identical ROMs.

I do see the diff errors that Bill reported, but have not investigated those.

I did notice an issue when there are multiple ROM targets being built in one run.  When the first ROM is built, several supporting binaries like romldr & dbgmon are built.  These binaries are dependent on build.inc which is customized for the platform being built.  When the next ROM is built, build.inc is updated as appropriate, but the supporting binaries are not rebuilt with it, so the supporting binaries are still for the original platform.  I solved the issue by adding dependencies on build.inc for these binaries (shown below).  However, I am not sure I did this the best way.

Thanks!

Wayne

wwarthen commented:
Oops, here is the snippet I added to Makefile in HBIOS directory:

```
romldr.bin:	romldr.asm build.inc
	$(TASM) romldr.asm romldr.bin

dbgmon.bin:	dbgmon.asm build.inc
	$(TASM) dbgmon.asm dbgmon.bin

```

cm68 commented:
no, that's actually the correct solution.   I found some others of these
when I built the N8 platform, but didn't catch these 2.

--curt

On Sun, Feb 23, 2020 at 3:23 PM Wayne Warthen <notifications@github.com>
wrote:

> Oops, here is the snippet I added to Makefile in HBIOS directory:
>
> romldr.bin:	romldr.asm build.inc
> 	$(TASM) romldr.asm romldr.bin
>
> dbgmon.bin:	dbgmon.asm build.inc
> 	$(TASM) dbgmon.asm dbgmon.bin
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wwarthen/RomWBW/pull/85?email_source=notifications&email_token=AJJUTK4HEBVAHSQA4ANQMSLREMAPJA5CNFSM4KZYD7DKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMWKPSA#issuecomment-590129096>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJJUTK5RKYAFFWU57CYTNIDREMAPJANCNFSM4KZYD7DA>
> .
>


-- 
*I *refine, *they* debug, *he/she* patches,* they* kludge.

cm68 commented:
the diff errors that Bill found are the result of files not existing in
both trees, and I changed the diff rule to handle this case.
I don't want to send you a bunch of pulls every time I find every bug, so I
tend to pile the fixes up. let me know if this getting to onerous.

--curt

On Sun, Feb 23, 2020 at 3:27 PM curt mayer <curt@zen-room.org> wrote:

> no, that's actually the correct solution.   I found some others of these
> when I built the N8 platform, but didn't catch these 2.
>
> --curt
>
> On Sun, Feb 23, 2020 at 3:23 PM Wayne Warthen <notifications@github.com>
> wrote:
>
>> Oops, here is the snippet I added to Makefile in HBIOS directory:
>>
>> romldr.bin:	romldr.asm build.inc
>> 	$(TASM) romldr.asm romldr.bin
>>
>> dbgmon.bin:	dbgmon.asm build.inc
>> 	$(TASM) dbgmon.asm dbgmon.bin
>>
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/wwarthen/RomWBW/pull/85?email_source=notifications&email_token=AJJUTK4HEBVAHSQA4ANQMSLREMAPJA5CNFSM4KZYD7DKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMWKPSA#issuecomment-590129096>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AJJUTK5RKYAFFWU57CYTNIDREMAPJANCNFSM4KZYD7DA>
>> .
>>
>
>
> --
> *I *refine, *they* debug, *he/she* patches,* they* kludge.
>
>

-- 
*I *refine, *they* debug, *he/she* patches,* they* kludge.

cm68 commented:
nascom also depends on I/O ports, so I'll add that dependency.

On Sun, Feb 23, 2020 at 3:31 PM curt mayer <curt@zen-room.org> wrote:

> the diff errors that Bill found are the result of files not existing in
> both trees, and I changed the diff rule to handle this case.
> I don't want to send you a bunch of pulls every time I find every bug, so
> I tend to pile the fixes up. let me know if this getting to onerous.
>
> --curt
>
> On Sun, Feb 23, 2020 at 3:27 PM curt mayer <curt@zen-room.org> wrote:
>
>> no, that's actually the correct solution.   I found some others of these
>> when I built the N8 platform, but didn't catch these 2.
>>
>> --curt
>>
>> On Sun, Feb 23, 2020 at 3:23 PM Wayne Warthen <notifications@github.com>
>> wrote:
>>
>>> Oops, here is the snippet I added to Makefile in HBIOS directory:
>>>
>>> romldr.bin:	romldr.asm build.inc
>>> 	$(TASM) romldr.asm romldr.bin
>>>
>>> dbgmon.bin:	dbgmon.asm build.inc
>>> 	$(TASM) dbgmon.asm dbgmon.bin
>>>
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/wwarthen/RomWBW/pull/85?email_source=notifications&email_token=AJJUTK4HEBVAHSQA4ANQMSLREMAPJA5CNFSM4KZYD7DKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMWKPSA#issuecomment-590129096>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AJJUTK5RKYAFFWU57CYTNIDREMAPJANCNFSM4KZYD7DA>
>>> .
>>>
>>
>>
>> --
>> *I *refine, *they* debug, *he/she* patches,* they* kludge.
>>
>>
>
> --
> *I *refine, *they* debug, *he/she* patches,* they* kludge.
>
>

-- 
*I *refine, *they* debug, *he/she* patches,* they* kludge.

wwarthen commented:
Not getting onerous at all.  yes, bundling the fixes is generally easiest.  Thanks!
